renku/service/cache/jobs.py

```diff
@@ -16,22 +16,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 """Renku service jobs management."""
+from datetime import datetime
 
 from renku.service.cache.base import BaseCache
+from renku.service.serializers.cache import UserJob
 
 
 class JobsManagementCache(BaseCache):
     """File management cache."""
 
     JOBS_SUFFIX = 'jobs'
+    schema = UserJob()
 
     def job_cache_key(self, user):
         """Construct cache key based on user and jobs suffix."""
         return '{0}_{1}'.format(user['user_id'], self.JOBS_SUFFIX)
 
-    def set_job(self, user, job_id, metadata):
+    def create_job(self, user, job):
+        """Create new user job."""
+        job['created_at'] = datetime.now()
+        self.set_job(user, job)
+
+    def set_job(self, user, job):
         """Cache job state under user hash set."""
-        self.set_record(self.job_cache_key(user), job_id, metadata)
+        job['updated_at'] = datetime.now()
+        job = self.schema.dump(job)
```

Comment:
Yes, I was thinking the same and looking to introduce the walrus as dependency for this. This would mean we should do the same on all our resources and change the data model slightly. Would be nice to have this as a followup since we would have to cover all of the service resources, not just jobs.